### DISCUSSION TITLE
vcpkg binary caching server?

### DISCUSSION THREAD

**Discussion by BullyWiiPlaza** [microsoft/vcpkg#31848] [Q&A]
Since building some ports is utterly horrible and takes hours to complete or might even get stuck (e.g. OpenCV, Boost or OpenSSL being "slow" ports to build), it would be great to have a central binary caching server to be able to download compiled ports instead of waiting very long to compile a simple port version update. It's not the first time that my vcpkg installation got "corrupted" after my PC ran out of memory while I tried to install port upgrades and ports wouldn't finish installing, even when I retried the installation despite deleting buildtrees/downloads etc., any ideas how to resolve this?

I researched and read [this](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching) article but I didn't find any server I could use. Is there one or are there plans to offer one? Maybe with GitHub Actions?

**Comment by michaelbprice** [comment]
There are currently no active plans to provide pre-built ports in the way you describe. The binary caching article you linked to is our current recommendation to reduce the amount of time spent compiling ports. We also added a GitHub Actions cache provider earlier this year if you are interested in speeding up your CI builds.

Pre-built ports is a bit tricky due to ABI complications that come with C++ libraries. That's not to say it's impossible or not valuable. There's probably a sweet spot to find, if there were enough demand for it.

**Comment by BullyWiiPlaza** [comment]
Sure, but why not just offer the latest MSVC, gcc and AppleClang compiler compiled versions of each port on each mainstream platform (Windows, Linux, MacOSX with x64, x86, static and dynamic binaries respectively/if applicable). This is what people would generally use when developing in C++ I suppose. For all additional needs, you can still fallback to compiling yourself. Basically just take all regression testing binary compilations from GitHub Actions/Microsoft Azure whatever and make them available to vcpkg users to significantly faster and more reliable port installations, if desired.

**Comment by xiaozhuai** [comment]
Vcpkg's caching policy is very strict. Even patch version changes to the compiler can cause cache missing. 
So only provide the binaries of the latest compiler version won't help much.
What our team is doing now:
1. Unified compiler version
2. Setup a CI server, build ports and put packages into a webdav server (vcpkg already support http binary caching source, s3 is also supported)
3. All the teamate should configure binary caching source as readonly, of course.
This is useful for teams, but may not be useful for individual developers.

**Reply by Osyotr** [reply]
This might help a little: https://github.com/microsoft/vcpkg/issues/26346#issuecomment-1319244766

**Reply by xiaozhuai** [reply]
@Osyotr Wow! Amazing, I didn't notice such an option. Thanks! I will give it a try.

**Reply by Neumann-A** [reply]
@Osyotr Do you keep a list of those helpful comments somewhere or do you just remember you wrote it at sometime and then search for it again?

**Reply by Osyotr** [reply]
@Neumann-A the latter :D
One day I'll update the FAQ...
